docs/examples/tutorial/clibraries/queue3.pyx

```diff
@@ -0,0 +1,61 @@
+# queue.pyx
+
+cimport cqueue
+
+cdef class Queue:
+    """A queue class for C integer values.
+
+    >>> q = Queue()
+    >>> q.append(5)
+    >>> q.peek()
+    5
+    >>> q.pop()
+    5
+    """
+    cdef cqueue.Queue* _c_queue
+    def __cinit__(self):
+        self._c_queue = cqueue.queue_new()
+        if self._c_queue is NULL:
+            raise MemoryError()
+
+    def __dealloc__(self):
+        if self._c_queue is not NULL:
+            cqueue.queue_free(self._c_queue)
+
+    cpdef append(self, int value):
+        if not cqueue.queue_push_tail(self._c_queue,
+                                      <void*> value):
+            raise MemoryError()
+
+    cdef extend(self, int* values, size_t count):
+        cdef size_t i
+        for i in range(count):
+            if not cqueue.queue_push_tail(
+                    self._c_queue, <void*> values[i]):
+                raise MemoryError()
+
+    # The `cpdef` feature is obviously not available for the `extend()`
+    # method, as the method signature is incompatible with Python argument
+    # types (Python doesn't have pointers). However, we can make a method
+    # called `extend_python` instead that accepts an arbitrary Python iterable.
+    cpdef extend_python(self, values):

```

Comment:
The naming of these two methods was intentional. The Python interface dominates the API of this class and should be consistent. `extend()` is the method name a Python user would expect. The additional C-interface should be a second level citizen wherever it cannot matche the Python interface.

Maybe we could rename the cdef `extend()` method to `extend_ints()`. That way, we could easily add an `extend_longs()` or `extend_chars()` method for convenience, as one would do in C APIs.